### ISSUE TITLE
Issue with cfssl-scan 

### ISSUE THREAD

**Issue by ptownbolt** [cloudflare/cfssl#758]
I'm running the latest code-base for cfssl-scan on OSX and Linux and get the following error when I scan some https sites like chase.com, wellsfargo.com. However when I scan google.com and cloudflare.com there is no problem. I debugged the code and the problem seems to on the line 290 of cfssl/scan/vendor/crypto/tls/handshake_client.go

285  switch certs[0].PublicKey.(type) {
286  case *rsa.PublicKey, *ecdsa.PublicKey:
287	 break
288	 default:
289		 c.sendAlert(alertUnsupportedCertificate)
290		 return fmt.Errorf("tls: server's certificate contains an unsupported type of public key: %T", 
 certs[0].PublicKey)
	}

$ cfssl scan -scanner=TLSDial  chase.com
Scanning chase.com...
=== chase.com ===
{
  "Connectivity": {
    "TLSDial": {
      "grade": "Bad",
      "error": "tls: server's certificate contains an unsupported type of public key: *rsa.PublicKey"
    }
  }
}

$ cfssl scan -scanner=TLSDial  cloudflare.com
Scanning cloudflare.com...
=== cloudflare.com ===
{
  "Connectivity": {
    "TLSDial": {
      "grade": "Good"
    }
  }
}

**Comment by umbernhard** [comment]
I'm having this same issue. I think it might be the way that the crypto package is vendored; it's out of date with its corresponding package in golang source. Trying to confirm that this is the issue. If it is, I think the right answer is to change the way the crypto package is used (un-vendor it, or use tris or something else).

**Comment by phpfs** [comment]
Hi there :)

Any updates on this?
Issue still exists for me with e.g. "yahoo.de" :(

Thanks so far!

**Comment by phpfs** [comment]
Updating dependencies might fix it - but will require a lot of changes in code :/

**Comment by elafargue** [comment]
Indeed, the cfssl code modified the system crypto package by adding its own hooks into the package, it would require taking those customizations out, but I am afraid the crypto package does not expose what the scan utility needs...

**Comment by koshatul** [comment]
Sorry, the PR won't fix this, just different issue with the crypto/tls package being overridden.
